### PR TITLE
fixes incorrect cint overflow in system

### LINKED ISSUE #22700: Error on syncio.nim
### Description

Error when trying to write a file bigger than 2147483647. 

when defined(windows):
  proc writeWindows(f: File; s: string; doRaise = false) =
    # Don't ask why but the 'printf' family of function is the only thing
    # that writes utf-8 strings reliably on Windows. At least on my Win 10
    # machine. We also enable setConsoleOutputCP(65001) now by default.
    # But we cannot call printf directly as the string might contain \0.
    # So we have to loop over all the sections separated by potential \0s.
    var i = c_fprintf(f, "%s", s)
    while i < s.len:
      if s[i] == '\0':
        let w = c_fputc('\0', f)
        if w != 0:
          if doRaise: raiseEIO("cannot write string to file")
          break
        inc i
      else:
        let w = c_fprintf(f, "%s", unsafeAddr s[i])
        if w <= 0:
          if doRaise: raiseEIO("cannot write string to file")
          break
        inc i, w

### Nim Version

2.0

### Current Output

```text
Error: unhandled exception: over- or underflow [OverflowDefect]
```

### Expected Output

_No response_

### Possible Solution

When changing 
var i = c_fprintf(f, "%s", s) to
var i:int64 c_fprintf(f, "%s", s)

the problem seems to be fixed

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ringabout**
fixes #22700

### CODE REVIEW

**Comment by Araq** [comment]
But fprintf returns `int`, https://cplusplus.com/reference/cstdio/fprintf/ so it cannot return the number of written bytes reliably for sizes beyond 2GB. What is really going on here?

**Comment by ringabout** [comment]
Yeah, as a return value from `c_fprintf`, it's totall fine. `i` is given a type of `cint`. But `i` is increased according to the length of the given strings, of which the length might exceed 2GB. 

In a nutshell, 
`i` is equal to `s.len`;  `i` of a `cint` type cannot handle a number bigger than 2147483647 and `s.len` can be more than 2GB.

```nim
    while i < s.len:
      if s[i] == '\0':
        let w = c_fputc('\0', f)
        if w != 0:
          if doRaise: raiseEIO("cannot write string to file")
          break
        inc i
      else:
        let w = c_fprintf(f, "%s", unsafeAddr s[i])
        if w <= 0:
          if doRaise: raiseEIO("cannot write string to file")
          break
        inc i, w # here
```

File: lib/std/syncio.nim [hunk 1]
```diff
@@ -246,7 +246,7 @@ when defined(windows):
     # machine. We also enable `setConsoleOutputCP(65001)` now by default.
     # But we cannot call printf directly as the string might contain \0.
     # So we have to loop over all the sections separated by potential \0s.
-    var i = c_fprintf(f, "%s", s)
+    var i = int c_fprintf(f, "%s", s)
```
Comment (ghost) [inline lib/std/syncio.nim:249]: won't using int mean that this will still fail on 32-bit systems for big files?

**Comment by Araq** [comment]
<strike>ok, but then you need to use `int64` here.</strike>

File: lib/std/syncio.nim (hunk 1 above)
Comment (ringabout) [reply lib/std/syncio.nim:249]:
```nim
var i = int c_fprintf(f, "%s", s)
while i < s.len:
```

Nope. `s.len` returns an `int` type, too and `i` is equal to `s.len`. It cannot exceed 2GB on 32-bit systems.

Comment (aftalavera) [reply lib/std/syncio.nim:249]: Did I help?

Comment (ringabout) [reply lib/std/syncio.nim:249]: You helped a bunch! Thank you!

**Comment by aftalavera** [comment]
Glad to hear it!

Im so proud, Its my first contribution!

**Pull request merged**